### DISCUSSION TITLE
The card inside  a card

### DISCUSSION THREAD

**Discussion by v479038280** [zauberzeug/nicegui#1295] [Ideas / Feature Requests]
If the card inside a card could have their own boader radius ?

like the card in the picture ,the card 2 and card 3 become rounded-none ,even if I changed their classes to rounded-full

![微信截图_20230731001904](https://github.com/zauberzeug/nicegui/assets/25096045/a0697b2d-05dc-4dcf-bbd9-3868a90f29b0)

**Comment by falkoschindler** [comment]
@v479038280:
1. Please show the code that you used to generate the screenshot. This might help others to understand what's going on.
2. Have a look at #726 where we discussed a similar issue with nested cards.
3. See https://nicegui.io/documentation/card#the_issue_with_nested_borders for related documentation.

The point is: Quasar's QCard, which `ui.card` is based on, manipulates borders of nested elements, because it is supposed to be used without padding and gaps. Since we already discussed this subject extensively in #726 and couldn't find a good solution that suits all needs, I don't think we will be able to change the behavior.

In your case I would simply wrap the content in another container like `ui.column`. Or you use `ui.column` _instead_  of `ui.card` and add border and/or shadow yourself as you like.

**Reply by v479038280** [reply]
Your answer is perfect. I'll try what you said.
here is the code. Just test the functionality of the component

```py
with ui.card().classes('bg-primary  flex-row w-full h-full 	rounded-full	'):
    with ui.card().classes('w-1/3 rounded-full'):
        ui.label('1')
    with ui.card().classes('w-2/3  rounded-full'):
        ui.label('2')
    with ui.card().classes('w-2/3'):
        ui.label('3')
    ui.label('4').classes('w-1/5 bg-secondary h-full')
```

**Comment by falkoschindler** [comment]
This problem has been resolved in [version 2.0.0](https://github.com/zauberzeug/nicegui/releases/tag/v2.0.0).
